### ISSUE TITLE
a11y (accessibility) of "open in new window" link icons

### ISSUE THREAD

**Issue by hchiam** [angular/angular#43512]
### Description

The "opens in new window" icon seems to not get announced correctly by a screen reader, or maybe not behave as you might expect: 
1. In Firefox, I hear "open_in_new", which gets announced as "open underscore in underscore new", instead of simply "opens in new window or tab". 
2. In Chrome, I don't hear that announced at all. 
3. When you on links with that icon, it does not seem to open in a new window or tab for me. 

I don't believe I have opening new windows/tabs automatically disabled, but hopefully someone can repro this bug to confirm. 

### What is the affected URL?

https://angular.io/docs

### Please provide the steps to reproduce the issue

1. Mac > Firefox > https://angular.io/docs > turn on VoiceOver > tab to the first link with that "opens in new window" icon. 
2. Mac > Chrome > https://angular.io/docs > turn on VoiceOver > tab to the first link with that "opens in new window" icon. 

Should get similar results for other screen readers. 

### Please provide the expected behavior vs the actual behavior you encountered

If the link is intended to behave as opening in a new tab/window, then ideally the announcement should match in Firefox with something like "open in new window or tab". 

If the link is intended to simply change the URL to a new page, then maybe the icon could simply be removed, since that icon usually indicates opening in a new tab/window. 

## Firefox:
<img width="1277" alt="firefox" src="https://user-images.githubusercontent.com/18131787/133946446-fa97d6f0-4218-43bd-a1b7-476b6b283c94.png">

## Chrome:
<img width="1103" alt="chrome" src="https://user-images.githubusercontent.com/18131787/133946450-88b4fc12-bd1e-4e96-8bd0-af5ac593c140.png">

## VoiceOver says "open underscore in underscore new" in Firefox:
<img width="949" alt="voiceover open_in_new" src="https://user-images.githubusercontent.com/18131787/133946453-bebe10de-dc7c-42d6-b598-a3e8f752da94.png">

**Comment by petebacondarwin** [comment]
We do not want the browser to open this link in a new tab/page.
We just wanted to use the icon to indicate that this page navigates outside of the current domain.
It is unfortunate that we have used Material Icons to provide this icon, for which we needed to add the content: "open_in_new". I believe we might be able to fix this by using the codepoint instead: `e89e`.

**Comment by petebacondarwin** [comment]
Or perhaps we just need to mark the icon with `aria-hidden="true"` but I am not sure how we could add this via CSS...

**Comment by petebacondarwin** [comment]
@hchiam - I don't have access to a screen reader to test. Can you try manually changing the CSS rule in the browser console to look like:  `content: "\e89e"` rather than `content: "open_in_new"`? If this stops the words from being read out then I can create a PR to implement this fix.

**Comment by crisbeto** [comment]
The correct thing to do here would be to add `aria-hidden="true"`. See the "Accessibility" section in the `mat-icon` docs: https://material.angular.io/components/icon/overview#accessibility

**Comment by petebacondarwin** [comment]
> The correct thing to do here would be to add `aria-hidden="true"`. See the "Accessibility" section in the `mat-icon` docs: https://material.angular.io/components/icon/overview#accessibility

@crisbeto - The trouble we have is that this is CSS-only. We are not able to mutate the HTML, we can only style it.

**Comment by hchiam** [comment]
> @hchiam - I don't have access to a screen reader to test. Can you try manually changing the CSS rule in the browser console to look like: `content: "\e89e"` rather than `content: "open_in_new"`? If this stops the words from being read out then I can create a PR to implement this fix.

@petebacondarwin - I can confirm that my screen reader doesn't read the words in the `content` when you use that codepoint. Looks good. Simple. 

That said, if we're stuck with a CSS-only fix, consider this: 

```css
a:not([target="_blank"])::after {
  content: "\e89e";
  content: "\e89e" / " (leaves angular.io) ";
  /* content repeated twice: first = backwards compatibility fallback; second = progressive enhancement */
  font-family: Material Icons;
}
```

or

```css
a:not([target="_blank"])::after {
  content: "\e89e";
  content: "\e89e"/ " (redirects to new domain) ";
  font-family: Material Icons;
}
```

So it'd say something like "link, latest standards (redirects to new domain)".

Live demo: https://codepen.io/hchiam/pen/KKqReQJ

The screen reader announces this in Chrome: 
<img width="455" alt="Screen reader output in Chrome" src="https://user-images.githubusercontent.com/18131787/134082142-af09ea73-e7cc-4920-af61-9ac456b792b0.png">

**Comment by petebacondarwin** [comment]
That is an interesting idea @hchiam. I think the wording should not mention redirects, since that is not what is happening.

Perhaps "leaves angular.io domain" or "external URL" might be accurate.

**Comment by gkalpak** [comment]
Nice! I didn't know `content` could accept alt text 🤯

**Comment by petebacondarwin** [comment]
It looks like "Alternative text after" is not well supported, specifically in FF and Safari: https://caniuse.com/mdn-css_properties_content_alt_text

So perhaps we should just go with the simple Unicode point fix?

**Comment by hchiam** [comment]
@petebacondarwin  
Agreed. I think we should just go with the simple codepoint. I recently remembered that we shouldn't rely on CSS-generated content to be read by screen readers anyways. https://developer.mozilla.org/en-US/docs/Web/CSS/content
